### PR TITLE
Remove calls to deprecated list_mbeds_ext functions

### PR DESCRIPTION
**Pull request by drewcassidy**
Replace all calls to `list_mbeds_by_targetid()` since it has been deprecated in mbed-ls. This should fix ARMmbed/mbed-ls#267

### CODE REVIEW

**Comment by drewcassidy** [comment]
...you ok there coveralls?

**Comment by drewcassidy** [comment]
Fixed title, since I apparently forgot what the name of the function was

File: mbed_host_tests/host_tests_plugins/host_test_plugins.py [hunk 1]
```diff
@@ -137,13 +137,15 @@ def check_mount_point_ready(self, destination_disk, init_delay=0.2, loop_delay=0
...
                 # Otherwise it will loop on same data.
                 mbeds = mbed_lstools.create()
-                mbeds_by_tid = mbeds.list_mbeds_by_targetid()   # key: target_id, value mbedls_dict()
-                if target_id in mbeds_by_tid:
-                    if 'mount_point' in mbeds_by_tid[target_id]:
-                        if mbeds_by_tid[target_id]['mount_point']:
-                            # Only assign if mount point is known (not None)
-                            new_destination_disk = mbeds_by_tid[target_id]['mount_point']
-                            break
+                mbed_list = mbeds.list_mbeds(unique_names=True, read_details_txt=True) #list of mbeds present
```
Comment (theotherjimmy) [inline mbed_host_tests/host_tests_plugins/host_test_plugins.py]: I don't think that you need the `read_details_text` and could use the default of False here. You're not using the `platform_unique_id`, so you can also drop `unique_names`.

File: mbed_host_tests/host_tests_plugins/host_test_plugins.py [hunk 2]
```diff
@@ -188,16 +190,18 @@ def check_serial_port_ready(self, serial_port, target_id=None, timeout=60):
...
-                if target_id in mbeds_by_tid:
-                    if 'serial_port' in mbeds_by_tid[target_id]:
-                        if mbeds_by_tid[target_id]['serial_port']:
-                            # Only assign if serial port is known (not None)
-                            new_serial_port = mbeds_by_tid[target_id]['serial_port']
-                            if new_serial_port != serial_port:
-                                # Serial port changed, update to new serial port from mbed-ls
-                                self.print_plugin_info("Serial port for tid='%s' changed from '%s' to '%s'..." % (target_id, serial_port, new_serial_port))
-                            break
+                mbed_list = mbeds.list_mbeds(unique_names=True, read_details_txt=True) #list of mbeds present
```
Comment (theotherjimmy) [inline mbed_host_tests/host_tests_plugins/host_test_plugins.py]: I don't think that you need the `read_details_text` and could use the default of False here. You're not using the `platform_unique_id`, so you can also drop `unique_names`.

**Comment by drewcassidy** [comment]
That should work, but I'm not in a position to test it right now

File: mbed_host_tests/host_tests_plugins/host_test_plugins.py (hunk 1 above)
Comment (drewcassidy) [reply mbed_host_tests/host_tests_plugins/host_test_plugins.py]: fixed in last commit

**Comment by jeromecoutant** [comment]
Hi
Could you release the version ?
Thx

**Comment by jeromecoutant** [comment]
When 1.3.1 release is planned ?,

Thx

**Comment by theotherjimmy** [comment]
@jeromecoutant Please open new issues for requests like this, I don't see all notifications on closed PRs/issues (I don't have them turned off, it may be a github issue). I'll do a release today or tomorrow, after I do an Mbed LS release.

**Pull request merged**